Gems/Atom/RHI/Vulkan/Code/Source/RHI.Builders/ShaderPlatformInterface.cpp

```diff
@@ -236,15 +225,7 @@ namespace AZ
                 AZ_Assert(false, "Invalid Shader stage.");
             }
 
-            AZStd::string prependFile;
-            if(platform.HasTag("mobile"))
-            {
-                prependFile = AndroidPlatformShaderHeader;
-            }
-            else
-            {
-                prependFile = WindowsPlatformShaderHeader;
-            }
+            AZStd::string prependFile{ AZ_TRAIT_ATOM_AZSL_PLATFORM_HEADER };
```

Comment:
That `platform` variable in this function is now unused.

It is triggering a compiler warning as an error in AR

```
 D:/workspace/o3de/Gems/Atom/RHI/Vulkan/Code/Source/RHI.Builders/ShaderPlatformInterface.cpp(153,50): warning C4100: 'platform': unreferenced formal parameter [D:\workspace\o3de\build\windows\External\Vulkan-3732ebb0\Code\Atom_RHI_Vulkan.Builders.Static.vcxproj]
10:33:14  D:/workspace/o3de/Gems/Atom/RHI/Vulkan/Code/Source/RHI.Builders/ShaderPlatformInterface.cpp(153,50): warning C4100:             const AssetBuilderSDK::PlatformInfo& platform, [D:\workspace\o3de\build\windows\External\Vulkan-3732ebb0\Code\Atom_RHI_Vulkan.Builders.Static.vcxproj]
10:33:14  D:/workspace/o3de/Gems/Atom/RHI/Vulkan/Code/Source/RHI.Builders/ShaderPlatformInterface.cpp(153,50): warning C4100:                                                  ^ [D:\workspace\o3de\build\windows\External\Vulkan-3732ebb0\Code\Atom_RHI_Vulkan.Builders.Static.vcxproj]
```